cmake/AddVSCode.cmake

```diff
@@ -0,0 +1,101 @@
+        file(READ ${args_FILE} current_launch_json)
+
+        string(JSON num_configs LENGTH ${current_launch_json} "configurations")
+        message("-- Found ${num_configs} VS Code Config(s) in .vscode/launch.json")
+        math(EXPR last_config "${num_configs} - 1")
+
+        # read recent build configuration from temporary launch configuraiton
+        file(READ ${PROJECT_BINARY_DIR}/launch.json.default default_launch_json)
+
+        # get generated default configuration
+        string(JSON default_launch_configuration GET ${default_launch_json} "configurations" 0)
+
+        # get the default configuration name
+        string(JSON default_launch_configuration_name GET ${default_launch_json} "configurations" 0 "name")
+
+        # get inputs required by default configuration
+        string(JSON default_launch_inputs GET ${default_launch_json} "inputs")
+
+        # iterate available configs and search for the default config
+        foreach(config_id RANGE ${last_config})
+            string(JSON config_name GET ${current_launch_json} "configurations" ${config_id} name)
+
+            # found Arterys default debug configuration
+            if (${config_name} STREQUAL ${default_launch_configuration_name})
+                message("-- Updating Artery default config in .vscode/launch.json")
+                set(default_config_found true)
+
+                # replace old config with newly generated one and save
+                string(JSON current_launch_json SET ${current_launch_json} "configurations" ${config_id} ${default_launch_configuration})
+                file(WRITE ${args_FILE} ${current_launch_json})
+            endif()
+        endforeach()
+
+        # launch.json is present but no default configuration was found
+        if (NOT ${default_config_found})
+            message("-- Append Artery default config to .vscode/launch.json")
+
+            # append default configuration
+            string(JSON current_launch_configuration SET ${current_launch_json} "configurations" ${num_configs} ${default_launch_configuration})
+
+            # append inputs to already existing inputs
+            string(JSON current_num_inputs LENGTH ${current_launch_configuration} "inputs")
+            string(JSON default_num_inputs LENGTH ${default_launch_inputs})
+            math(EXPR last_default_input "${default_num_inputs} - 1")
+
+            foreach(default_input_index RANGE ${last_default_input})
+                string(JSON input GET ${default_launch_inputs} ${default_input_index})
+                math(EXPR current_input_index "${current_num_inputs} + ${default_input_index}")
+                string(JSON current_launch_configuration SET ${current_launch_configuration} "inputs" ${current_input_index} ${input} )
+            endforeach()
+
+            file(WRITE ${args_FILE} ${current_launch_configuration})
+        endif()
+    else()
+        message("-- Generating default debug config for VS Code")
```

Comment:
If no *.vscode* directory exists at all then only the build directory is touched, right?

Reply:
`cmake` will create `.vscode` and `.vscode/launch.json`, when it is told to by the user, explicitly enabling `VSCODE_LAUNCH_INTEGRATION`. By default, nothing is touched and the project directory remains clean :)